src/shogun/lib/SGMatrix.h

```diff
@@ -130,6 +131,15 @@ template<class T> class SGMatrix : public SGReferencedData
 		virtual ~SGMatrix();
 
 #ifndef SWIG // SWIG should skip this part
+		/** Resize matrix
+		 * If the new matrix is larger, fill the new memory with 0.
+		 *
+		 * @param rows new row numbers
+		 * @param cols new column numbers
+		 * @return if resizing was successful
+		 */
+		bool resize_matrix(index_t rows, index_t cols);
+
 		/** Get a column vector
```

Comment:
@karlnapf The dimensions of the result matrix are determined by whether the input matrices are transposed. Currently there is no method to set matrices' rows and cols after declaration `SGMatrix<T> mat;`.

Now I am trying to do:
```
SGMatrix<T> result;
if (transposed_A)
    result.resize(rows, cols);
if (transposed_B)
    result.resize(rows, cols);
else
    result.resize(rows, cols);

if (A.on_gpu())
     result = to_gpu(result);
matrix_prod(A, B, result, transpose_A, transpose_B);
```
Without the method, the codes will be like:
```
if (transposed_A)
    result.resize(rows, cols);
    if (A.on_gpu())
         result = to_gpu(result);
    matrix_prod(A, B, result, transpose_A, transpose_B);
if (transposed_B)
    result.resize(rows, cols);
    if (A.on_gpu())
         result = to_gpu(result);
    matrix_prod(A, B, result, transpose_A, transpose_B);
else
    result.resize(rows, cols);
    if (A.on_gpu())
         result = to_gpu(result);
     matrix_prod(A, B, result, transpose_A, transpose_B);
```